Library/Homebrew/dev-cmd/create.rb

```diff
@@ -141,20 +141,19 @@ def create_cask(args:)
 
   def create_formula(args:)
     fc = FormulaCreator.new(args)
+    fc.url = args.named.first
     fc.name = if args.set_name.blank?
-      stem = Pathname.new(args.named.first).stem.rpartition("=").last
-      print "Formula name [#{stem}]: "
-      __gets || stem
+      name = url2name(fc.url)
+      print "Formula name [#{name}]: "
+      __gets || name
     else
       args.set_name
     end
-    fc.version = args.set_version
+    fc.version = args.set_version if args.set_version
     fc.license = args.set_license
```

Comment:
Given `args` is already passed to `FormulaCreator.new`: could consider moving this logic in `initialize` in there instead.

Reply:
I try to do this in a separate PR.

Reply:
`args` to `initialize` were done in https://github.com/Homebrew/brew/pull/16260